src/components/App.jsx

```diff
@@ -1,75 +1,180 @@
+      content: intl.formatMessage(messages['siteheader.links.schools']),
+    },
+  ];
+  const userMenu = [
+    {
+      type: 'item',
+      href: `${process.env.LMS_BASE_URL}`,
+      content: intl.formatMessage(messages['siteheader.user.menu.dashboard']),
+    },
+    {
+      type: 'item',
+      href: `${process.env.BASE_URL}/u/${username}`,
+      content: intl.formatMessage(messages['siteheader.user.menu.profile']),
+    },
+    {
+      type: 'item',
+      href: `${process.env.LMS_BASE_URL}/account/settings`,
+      content: intl.formatMessage(messages['siteheader.user.menu.account.settings']),
+    },
+    {
+      type: 'item',
+      href: process.env.LOGOUT_URL,
+      content: intl.formatMessage(messages['siteheader.user.menu.logout']),
+    },
+  ];
+  const loggedOutItems = [
+    {
+      type: 'item',
+      href: `${process.env.LMS_BASE_URL}/login`,
+      content: intl.formatMessage(messages['siteheader.user.menu.login']),
+    },
+    {
+      type: 'item',
+      href: `${process.env.LMS_BASE_URL}/register`,
+      content: intl.formatMessage(messages['siteheader.user.menu.register']),
+    },
+  ];
 
-    return (
-      <div>
-        <SiteHeader />
-        <main>
-          <Switch>
-            <Route path="/u/:username" component={ConnectedProfilePage} />
-            <Route path="/error" component={ErrorPage} />
-            <Route path="/notfound" component={NotFoundPage} />
-            <Route path="*" component={NotFoundPage} />
-          </Switch>
-        </main>
-        <SiteFooter
-          siteName={process.env.SITE_NAME}
-          siteLogo={FooterLogo}
-          marketingSiteBaseUrl={process.env.MARKETING_SITE_BASE_URL}
-          supportUrl={process.env.SUPPORT_URL}
-          contactUrl={process.env.CONTACT_URL}
-          openSourceUrl={process.env.OPEN_SOURCE_URL}
-          termsOfServiceUrl={process.env.TERMS_OF_SERVICE_URL}
-          privacyPolicyUrl={process.env.PRIVACY_POLICY_URL}
-          facebookUrl={process.env.FACEBOOK_URL}
-          twitterUrl={process.env.TWITTER_URL}
-          youTubeUrl={process.env.YOU_TUBE_URL}
-          linkedInUrl={process.env.LINKED_IN_URL}
-          googlePlusUrl={process.env.GOOGLE_PLUS_URL}
-          redditUrl={process.env.REDDIT_URL}
-          appleAppStoreUrl={process.env.APPLE_APP_STORE_URL}
-          googlePlayUrl={process.env.GOOGLE_PLAY_URL}
-          handleAllTrackEvents={sendTrackEvent}
-        />
-      </div>
-    );
+  return (
+    <div>
+      <SiteHeader
+        logo={HeaderLogo}
+        loggedIn
+        username={username}
+        avatar={avatar}
+        logoAltText={configuration.SITE_NAME}
+        logoDestination={configuration.MARKETING_SITE_BASE_URL}
+        mainMenu={mainMenu}
+        userMenu={userMenu}
+        loggedOutItems={loggedOutItems}
+      />
+      <main>
+        <Switch>
+          <Route path="/u/:username" component={ConnectedProfilePage} />
```

Comment:
In a larger MFE, and if the reusable organisms were actually in their own repos/own npm packages, you could imagine replacing this statically-defined `<Switch>` with something that discovers the pages it wants to include.  Plugin-architecture-style.  Doing that would help make App.jsx even more de-coupled from the actual content of a particular MFE, and thus more reusable.